File: redo/reredo/darwin/datetimepicker.m
// 14 august 2015
#import "uipriv_darwin.h"

struct uiDateTimePicker {
	uiDarwinControl c;
	NSDatePicker *dp;
};

uiDarwinDefineControl(
	uiDateTimePicker,						// type name
	uiDaateTimePickerType,					// type function
	dp									// handle
)

static uiDateTimePicker *finishNewDateTimePicker(NSDatePickerElementFlags elements)
{
	uiDateTimePicker *d;

	d = (uiDateTimePicker *) uiNewControl(uiDateTimePickerType());

	d->dp = [[NSDatePicker alloc] initWithFrame:NSZeroRect];
	// TODO text field stuff
	[d->dp setDatePickerStyle:NSTextFieldAndStepperDatePickerStyle];
	[d->dp setDatePickerElements:elements];
	[d->dp setDatePickerMode:NSSingleDateMode];
	// TODO get date picker font

	uiDarwinFinishNewControl(d, uiDateTimePicker);

	return d;
}

uiDateTimePicker *uiNewDateTimePicker(void)
{
	return finishNewDateTimePicker(NSYearMonthDayDatePickerElementFlag | NSHourMinuteSecondDatePickerElementFlag);
}

uiDateTimePicker *uiNewDatePicker(void)
{
	return finishNewDateTimePicker(NSYearMonthDayDatePickerElementFlag);
}

uiDateTimePicker *uiNewTimePicker(void)
{
	return finishNewDateTimePicker(NSHourMinuteSecondDatePickerElementFlag);
}

File: wpf/datetimepicker.cpp
// 26 november 2015
#include "uipriv_wpf.hpp"

struct uiDateTimePicker {
	uiWindowsControl c;
	DUMMY dummy;
};

uiWindowsDefineControl(
	uiDateTimePicker,						// type name
	uiDateTimePickerType,					// type function
	dummy								// handle
)

static uiDateTimePicker *finishNewDateTimePicker(void)
{
	uiDateTimePicker *d;

	d = (uiDateTimePicker *) uiNewControl(uiDateTimePickerType());

	d->dummy = mkdummy(L"uiDateTimePicker");

	uiWindowsFinishNewControl(d, uiDateTimePicker, dummy);

	return d;
}

uiDateTimePicker *uiNewDateTimePicker(void)
{
	return finishNewDateTimePicker();
}

uiDateTimePicker *uiNewDatePicker(void)
{
	return finishNewDateTimePicker();
}

uiDateTimePicker *uiNewTimePicker(void)
{
	return finishNewDateTimePicker();
}
